src/lib/structures/expr.ml

```diff
@@ -1210,10 +1210,20 @@ let mk_nary_eq l =
   with Exit ->
     vrai
 
-let mk_distinct ~iff tl =
-  match tl with
-  | [a; b] -> neg (mk_eq ~iff a b)
-  | _ -> neg (mk_nary_eq tl)
+let mk_distinct ~iff args =
+  (* This fix makes sure that the smart constructor agrees with
+     the SMT-LIB specification when used with at least 3 arguments.
+     To prevent a soundness bug, we translate the expected expression into a
+     conjonction of binary disequations. *)
```

Comment:
Note: this is not specifically related to the SMT-LIB specification but to the usual definition of `distinct`.

Reply:
Also the comment should point to #889 which will track the fix in the `next` branch

Reply:
Sorry, I maybe was not clear. #889 is not about this patch, #889 is about the issue with `distinct`. We will fix the issue with `distinct` on `next` in a different way, which is fine, but we should still refer to it here.

Reply:
Ok :)